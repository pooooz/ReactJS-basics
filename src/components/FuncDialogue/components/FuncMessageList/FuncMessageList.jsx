import React from 'react';
import styles from './FuncMessageList.module.scss';
import { FuncMessage } from './components/FuncMessage/FuncMessage';

const FuncMessagePure = React.memo(FuncMessage);

export const FuncMessageList = (props) => {
  return (
    <div className={styles.message__list}>
      {props.messages.map((message, idx) => (
        <FuncMessagePure
          message={message.text}
          author={message.author}
          key={idx}
        />
      ))}
    </div>
  );
};
