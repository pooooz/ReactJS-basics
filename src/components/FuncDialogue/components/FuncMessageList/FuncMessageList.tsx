import React, { FC } from 'react';
import styles from './FuncMessageList.module.scss';
import { FuncMessage } from './components/FuncMessage/FuncMessage';
import { Message } from '../../FuncDialogue';

interface FuncMessageListProps {
  messages: Message[];
}

const FuncMessagePure = React.memo(FuncMessage);

export const FuncMessageList: FC<FuncMessageListProps> = (props) => {
  return (
    <div className={styles.message__list}>
      {props.messages.map((message, idx) => (
        <FuncMessagePure
          message={message.text}
          author={message.author}
          key={idx}
        />
      ))}
    </div>
  );
};
