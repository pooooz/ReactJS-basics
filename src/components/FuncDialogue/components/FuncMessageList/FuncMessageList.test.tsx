import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { FuncMessageList } from './FuncMessageList';

describe('FuncMessageList', () => {
  it('Render component', () => {
    render(
      <FuncMessageList messages={[{ text: 'Render', author: 'Render' }]} />
    );
  });

  it('Render with snapshot', () => {
    const { asFragment } = render(
      <FuncMessageList messages={[{ text: 'Render', author: 'Render' }]} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Render with text', () => {
    render(
      <FuncMessageList messages={[{ text: 'Render', author: 'Render' }]} />
    );
    expect(screen.getByText(/Author: <\w*>/));
  });
});
