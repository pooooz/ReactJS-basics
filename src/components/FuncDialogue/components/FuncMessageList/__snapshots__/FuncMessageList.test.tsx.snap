// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FuncMessageList Render with snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <p
        data-text="Render"
      >
        Render
      </p>
      <p
        data-text="Author: <Render>"
      >
        Author: &lt;Render&gt;
      </p>
    </div>
  </div>
</DocumentFragment>
`;
